radix-engine-stores/src/rocks_db_with_merkle_tree/mod.rs

```diff
@@ -216,7 +220,8 @@ impl ReadableTreeStore for RocksDBWithMerkleTreeSubstateStore {
         self.db
             .get_cf(self.cf(MERKLE_NODES_CF), &encode_key(key))
             .unwrap()
-            .map(|bytes| scrypto_decode(&bytes).unwrap())
+            .map(|bytes| scrypto_decode::<VersionedTreeNode>(&bytes).unwrap())
+            .map(|versioned| versioned.into_latest())
```

Comment:
Technically, the test-only rocksdb impl in Engine repo does not need this, but I wanted the reviewers to see how I plan to support that in the Node repo.